_posts/2022-05-17-noise.md

```diff
@@ -0,0 +1,333 @@
+
+Within this handshake, Alice and Bob reciprocally authenticate their static keys `s` using ephemeral keys `e`. We note that while Bob's static key is assumed to be known to Alice (and hence is not transmitted), Alice's static key is sent to Bob encrypted with a key derived from both parties ephemeral keys and Bob's static key.
+
+**Security considerations on identity-hiding (active attacker)**: Alice's static key is encrypted with forward secrecy to an authenticated party. An active attacker initiating the handshake can check candidates for Bob's static key against recorded/accepted exchanged handshake messages.
+
+
+### The `XX` and `XXpsk0` Handshakes
+
+If Alice is not aware of any static key belonging to Bob (and neither Bob knows anything about Alice), she can execute an `XX` handshake, where each party tran**X**mits to the other its own static key. 
+
+The handshake goes as follows:
+
+```
+ XX:
+    ->  e
+    <-  e, ee, s, es
+    ->  s, se
+```
+
+We note that the main difference with `XK1` is that in second step Bob sends to Alice his own static key encrypted with a key obtained from an ephemeral-ephemeral Diffie-Hellman exchange.
+
+
+This handshake can be slightly changed in case both Alice and Bob pre-shares some secret `psk` which can be used to strengthen their mutual authentication during the handshake execution. One of the resulting protocol, called `XXpsk0`, goes as follow:
+
+```
+ XXpsk0:
+    ->  psk, e
+    <-  e, ee, s, es
+    ->  s, se
+```
+The main difference with `XX` is that Alice's and Bob's static keys, when transmitted, would be encrypted with a key derived from `psk` as well.
+
+
+**Security considerations on identity-hiding (active attacker)**: Alice's static key is encrypted with forward secrecy to an authenticated party for both `XX` and `XXpsk0` handshakes. In `XX`, Bob's static key is encrypted with forward secrecy but is transmitted to a non-authenticated user which can then be an active attacker. In `XXpsk0`, instead, Bob's secret key is protected by forward secrecy to a partially authenticated party (through the pre-shared secret `psk` but not through any static key), provided that `psk` was not previously compromised (in such case identity-hiding properties provided by the `XX` handshake applies).
+
+# Session Management and Multi-Device Support
+
+When two users complete a Noise handshake, an encryption/decryption session - or _Noise session_ - consisting of two Cipher States is instantiated.
+
+By identifying Noise session with a `session-id` derived from the handshake's cryptographic material, we can take advantage of the [PubSub/GossipSub](https://github.com/libp2p/specs/tree/master/pubsub) protocols used by Waku for relaying messages in order to manage instantiated Noise sessions.
+
+The core idea is to exchange after-handshake messages  (encrypted with a Cipher State specific to the Noise session), over a content topic derived from the (secret) `session-id` the corresponding session refers to.
```

Comment:
Would this be done in a way that makes the specific application still apparent in the content topic? We may have to think about how the content topic should be used here - different applications generally require to be able to `filter` that application based on the content topic.

Reply:
According to the (under review) Noise session management [raw RFC](https://github.com/vacp2p/rfc/pull/504), the content topic would be `/wakunoise/1/<ct-id>/proto`, with `ct-id` derived from `session-id`: so it is apparent the use of Noise, but not immediately apparent the application using Noise for key-exchange. We may address this later when the RFC become at least draft and we have a better understanding on how managing sessions. What do you think?

Reply:
Makes sense.

Reply:
Related comment https://github.com/vacp2p/rfc/pull/504#discussion_r880283108